src/Struct/FindologicProduct.php

```diff
@@ -1000,4 +981,24 @@ protected function isApiIntegration(): bool
     {
         return $this->config->getIntegrationType() === IntegrationType::API;
     }
+
+    protected function addCategoryNamesRecursively(string $categoryPath, array $categories): array
+    {
+        $parentCategory = explode('_', $categoryPath);
+
+        return array_merge($categories, $parentCategory);
+    }
+
+    /**
+     * For API Integrations, we have to remove special characters from the attribute key as a requirement for
+     * sending data via API.
+     */
+    private function processAttributeKey(?string $key): ?string
```

Comment:
* Calling something `process` isn't really telling anything, and it also does not matter what "it" really processes. How about `getAttributeKey`. The internals do not really matter for anyone, except the one really interested in what it does and then the method name helps understanding.
* Change the visibility from `private` to `protected`, as otherwise extensions can not call it.